pkgs/development/python-modules/hikari/default.nix

```diff
@@ -0,0 +1,77 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pytestCheckHook
+, pythonOlder
+, pythonRelaxDepsHook
+, pytest-runner
+, aiohttp
+, attrs
+, multidict
+, colorlog
+, pynacl
+, git
+, pytest
+, pytest-cov
+, pytest-randomly
+, pytest-asyncio
+, mock
+}:
+
+buildPythonPackage rec {
+  pname = "hikari";
+  version = "2.0.0.dev119";
+
+  src = fetchFromGitHub {
+    owner = "hikari-py";
+    repo = "hikari";
+    rev = version;
+    hash = "sha256-4uMY0NZaLxl52Wl471xQ6JqWWVW5QzMOyrXeVaZltfo=";
+    leaveDotGit = true;
```

Comment:
I have some doubts to use leaveDotGit, as it may break on changes upstream as the .git seems to be non deterministic (see https://github.com/NixOS/nixpkgs/issues/8567#issuecomment-1140237191 ) . As far as I can see you are just setting an environment variable with the latest git Sha from the log.
I personally would try to "hardcore" that to the specific Sha of the version (and leave a comment that both need to be updated)
I haven't tested if there are other git commands involved in the build.
Btw: thanks for packaging this, i got an older version in a personal (private) flake and forgot to upstream that. 🙂

Reply:
> [the] best solution would be to have an update script that updates the version and the Sha

Correct me if I'm wrong, but can't update scripts only return a version (no SHAs or other data).

Reply:
We can freely prepare an `update.sh` in which we get and update "sha1".

Reply:
[nix-update](https://github.com/Mic92/nix-update) should work fine here.

Reply:
I don't see how nix-update can update a custom string aside the rev, hash and the packagers strings. Could you point to the right parameters? 

The issue here is to store the Sha beside the version tag, so replace it in the `about.py` so that there is no dependency to `leaveDotGit` as it may break on pushes.
Another option could be ofc to not use version tag, use the tagged Sha and use this in the about.py too, then a comment to always use Shas would also be enough.

Reply:
Oh I somehow missed that :see_no_evil:

Reply:
I've removed `leaveDotGit` in favor of manually specifying the Git hash.